test/src/e2e_vm_tests/test_programs/should_fail/non_payable_non_zero_coins_let_binding/src/main.sw

```diff
@@ -0,0 +1,23 @@
+script;
+
+dep wallet_abi;
+
+use std::constants::ZERO_B256;
+use wallet_abi::Wallet;
+
+fn main() {
+    let contract_address = 0x9299da6c73e6dc03eeabcce242bb347de3f5f56cd1c70926d76526d7ed199b8b;
+    let caller = abi(Wallet, contract_address);
+    let amount_to_send = 200;
+    let recipient_address = Address::from(0x9299da6c73e6dc03eeabcce242bb347de3f5f56cd1c70926d76526d7ed199b8b);
+    // calling a non-payable method with a non-zero amount `coins`
+    // this must result in an error
+    // we use a chain of let-bindings that ends in a non-zero literal
+    let fortytwo = 42;
```

Comment:
Can we also check that if this values is 0, then we don't emit an error? This would prove that the despite the fact that the number of coins is a _variable_, we're still able to deduce that it's actually just a 0 and not conservatively emit an error.

Reply:
Done in https://github.com/FuelLabs/sway/pull/3450/commits/c4065f8e326f6755a128aedbc7418100705c98c7